micro-rdk-installer/src/main.rs

```diff
@@ -92,13 +105,39 @@ fn create_nvs_partition_binary(config_path: String, size: usize) -> Result<Vec<u
     Ok(NVSPartitionData::try_from(part)?.to_bytes())
 }
 
+fn write_credentials_to_app_binary(
+    binary_path: &str,
+    nvs_data: &[u8],
+    nvs_size: u64,
+    nvs_start_address: u64,
+) -> Result<(), Error> {
+    let mut app_file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .open(binary_path)
+        .map_err(Error::FileError)?;
+    let file_len = app_file.metadata().map_err(Error::FileError)?.len();
+    if (file_len - nvs_start_address) <= nvs_size {
```

Comment:
this check is a bit dubious? you would really need the info from the partition table to make sure the NVS partition doesn't override others partition